CsharpExtrasTest/Extensions/DictionaryTest.cs

```diff
@@ -63,6 +63,25 @@ public void GivenStringIntDictionaryWhenValuesMappedToStringThenKeysRemainUnchan
             Assert.AreEqual(mockDict.Keys, mappedDictionary.Keys);
         }
 
+        [Test]
+        [Category("Unit")]
+        public void GivenDictionaryWhenAddWithKeyDerivedFromValueCalledThenCorrectItemAddedToDictionary()
+        {
+            //Arrange
+            IDictionary<string, int> mockDict = MockStrIntDictionary();
+            string KeyFromValue(int value)
+            {
+                return value + "";
+            }
+
+            //Act
+            mockDict.AddWithKeyDerivedFromValue(5, KeyFromValue);
+            //Assert
+
+            Assert.IsTrue(mockDict.ContainsKey("5"));
+            Assert.IsTrue(mockDict.Values.Contains(5));

```

Comment:
Add an assertion that the value mapped from "5" is indeed 5.

Reply:
I would also add an assert that the dictionary only contains one item.